src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.put(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded at "
+                    + SupplyOperation.class);
+        }
+    }
```

Comment:
The `SupplyOperation` class should not instantiate a new `Storage` object. Instead, it should use a single instance of `Storage` that is shared across all operations. This ensures that all operations are working with the same data. Consider passing the `Storage` instance to the `SupplyOperation` via the constructor or method parameter.